galicaster/mediapackage/mediapackage.py

```diff
@@ -1520,26 +1525,39 @@ def getOCCaptureAgentProperties(self):
 
     def setProperty(self, prop=None, value=None):
         """Sets the value of a property in the mediapackage.
+        The enqueue_params property will be JSON serialized.
         Args:
             prop (str): the name of the property.
-            value (str): the new value of the property name.
+            value (str): the new value of the property.
         """
         if not prop or value is None:
             return None
-        else:
-            self.properties[prop] = value
-            return True
+        property_value = value
+        if prop == "enqueue_params":
```

Comment:
To allow for this kind of API:
`media_package.setProperty('enqueue_params', {'workflow': 'test'})`
What we need to do is to support serializing/deserializing objects on the serializer/deserializer, not here.

Also, this support should be added generically, not just for an "enqueue_params" property.

Essentially, to add this functionality what we should do is to support dictionary properties, at least. And serialize/deserialize them on the corresponding classes. The setProperty function should not be the one carrying this logic